src/page-multi-facility/ReferenceDataModal/SyncNewFacility.tsx

```diff
@@ -42,7 +42,7 @@ const SyncNewFacility: React.FC<Props> = ({ facilityId, onClose }) => {
     mappedReferenceFacilities,
     referenceFacilities,
   );
-  if (!facilityId) return null;
+  if (!facilityId || isEmpty(unmappedReferenceFacilities)) return null;
```

Comment:
I think you meant to add this to `SyncNewReferenceData`, right?

Reply:
Or you can add it as part of the `newlyAddedReferenceData` boolean check

Reply:
Hmm..good question...I initially didn't add it for `SyncNewReferenceData` because I was thinking that any "newly added" reference facilities that have not been displayed to the user yet would not be synced yet. I'm not sure now if that's a solid assumption. I added it to this modal because this would open whenever the user saves a new facility, and we wouldn't want to show it if all of the reference facilities are already synced.

Reply:
Yup, I think this makes sense to include here.

I'm also confusing myself about what the exact conditions are for showing the `SyncNewReferenceData` in the context of handling both mapping unmapped user facilities to new reference facilities and/or adding new reference facilities as user facilities when all of the existing user facilities are already mapped to a reference facility.